zerver/openapi/zulip.yaml

```diff
@@ -6892,35 +6896,48 @@ paths:
-
-        **Changes**: Before Zulip 7.0 (feature level 172), anyone could add
-        a topic to messages without topic regardless of topic edit
-        permissions. Now topic editing restrictions apply to messages without a
-        topic as well.
-
-        Before Zulip 7.0 (feature level 172), users were allowed to move messages
-        that had passed the time limit by using `change_all` value for `propagate_mode`
-        parameter. Now, the server returns an error if users, other than organization
-        administrators and moderators, try to move messages that have passed the time
-        limit.
+        See [configuring message editing][config-message-editing] for detailed
+        documentation on when users are allowed to edit message content and
+        [restricting moving messages][restrict-move-messages] for detailed
+        documentation on when users are allowed to change a message's topic
+        and/or stream.
+
+        **Changes**: Prior to Zulip 7.0 (feature level 172), anyone could add a
+        topic to stream messages without a topic, regardless of the organization's
+        [topic editing permissions](/help/restrict-moving-messages). As of this
+        feature level, messages without topics have the same restrictions for
+        topic edits as messages with topics.
+
+        Before Zulip 7.0 (feature level 172), by using the `change_all` value for
+        the `propagate_mode` parameter, users could move messages after the
+        organization's configured time limits for changing a message's topic or
+        stream had passed. As of this feature level, the server will [return an
+        error](/api/update-message#response) with `"code":
+        "MOVE_MESSAGES_TIME_LIMIT_EXCEEDED"` if users, other than organization
+        administrators or moderators, try to move messages after these time
+        limits have passed.
+
+        Before Zulip 7.0 (feature level 162), users who were not administrators or
+        moderators could only edit topics if the target message was sent within the
+        last 3 days. As of this feature level, that time limit is now controlled by
+        the realm setting `move_messages_within_stream_limit_seconds`. Also at this
+        feature level, a similar time limit for moving messages between streams was
+        added, controlled by the realm setting
+        `move_messages_between_streams_limit_seconds`. Previously, all users who
+        had permission to move messages between streams did not have any time limit
+        restrictions when doing so.
+
+        As of Zulip 7.0 (feature level 159), the realm settings
+        `move_messages_between_streams_policy` and `edit_topic_policy` always
+        determine if a user has permission to move messages between streams or edit
+        topics. Previously, editing streams and topics of messages was forbidden
+        if `allow_message_editing` was `false`, regardless of the above policies.
```

Comment:
Here also, I think putting it in "Before" form (like in the existing text) rather than "As of" form ends up being clearer.

In addition to the general points in the other comment, the "always" here makes me a bit nervous because it says there's no *other* constraints that apply.

Reply:
I had removed that "always" and must have not pushed that most recent version. The current revisions have gone back to the "Before" form for this note, though I did expand the text to note these are realm settings.